### ISSUE TITLE
2.27.0: cannot type capital letters into commit message in vi

### ISSUE THREAD

**Issue by dmcbride** [git-for-windows/git#2758]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.27.0.windows.1
cpu: x86_64
built from commit: 907ab1011dce9112700498e034b974ba60f8b407
sizeof-long: 4
sizeof-size_t: 8
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver
Microsoft Windows [Version 10.0.18363.959]
(c) 2019 Microsoft Corporation. All rights reserved.
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
$ cat /etc/install-options.txt

Editor Option: VIM
Custom Editor Path:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Rebase
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

Using a vimrc file that does nothing but source a local copy of https://github.com/dmcbride/git-tools/blob/master/vim/.vimrc 

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git commit
```

 - What did you expect to occur after running these commands?

Try to type a commit message involving capital letters, everything appears exactly as I type.

 - What actually happened instead?

Try to type a commit message involving capital letters, but they don't show up. The lower case version of those letters doesn't show up either, literally nothing appears to happen.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

I haven't noticed it be specific to a repository. This also did not happen with 2.26. I tried with and without pseudo console support enabled, no difference.

Note that editing any other file with vi from bash works fine, just not the COMMIT_EDITMSG (so far), and only when run from under git (if I edit the commit message from another git bash window, then it works fine, all the capital letters).

**Comment by dpells** [comment]
> - What did you expect to occur after running these commands?
Try to type a commit message involving capital letters, everything appears exactly as I type.
> - What actually happened instead?
Try to type a commit message involving capital letters, but they don't show up. The lower case version of those letters doesn't show up either, literally nothing appears to happen.

I'm also seeing it with commands in normal mode, specifically `I` and `A`. Instead of inserting at the beginning or appending to the end of the line, there's no action

**Comment by gitting** [comment]
This has been an issue for me as well. Had to go back to 2.26 to be able to use `vim` as an editor.

**Comment by rimrul** [comment]
Is this still an issue with 2.28.0? because this sounds a lot like #2689 and the users in that issue reported, that the snapshots fixed it.

**Comment by dpells** [comment]
Fixed for me on 2.28.0. I can type capitals in visual and normal mode

**Comment by gitting** [comment]
I used 2.28 rc2 and the issue was not there. This release works too. Thanks.
